docs/developer-docs/latest/update-migration-guides/migration-guides/v4/code/backend/routes.md

```diff
@@ -0,0 +1,88 @@
+---
+title: v4 code migration - Routes - Strapi Developer Docs
+description: Migrate routes of a Strapi application from v3.6.8 to v4.0.x
+canonicalUrl:  http://docs.strapi.io/developer-docs/latest/update-migration-guides/migration-guides/v4/code/backend/routes.html
+---
+
+# v4 code migration: Updating routes
+
+!!!include(developer-docs/latest/update-migration-guides/migration-guides/v4/snippets/code-migration-intro.md)!!!
+
+::: strapi v3/v4 comparison
+In both Strapi v3 and v4, creating content-types automatically generates core API routes.
+
+In Strapi v3, routes are defined in JSON files that export an object containing configurations.
+
+In Strapi v4, routes are defined in JavaScript files, called router files, that export the result of a call to the [`createCoreRouter` factory function](/developer-docs/latest/development/backend-customization/routes.md#configuring-core-routers), with or without further customization.
+:::
+
+Migrating [routes](/developer-docs/latest/development/backend-customization/routes.md) to Strapi v4 consists in making sure that each router file uses the `createCoreRouter` factory function introduced in v4.
+
+Due to the differences between routes implementation in Strapi v3 and v4, it's recommended to create a new router file, then optionally bring existing v3 customizations into the new file and adapt them when necessary.
+
+A new router file can be created:
+
+- manually, starting at step 1 of the following procedure,
+- or automatically, using the [`strapi generate` interactive CLI](/developer-docs/latest/developer-resources/cli/CLI.md#strapi-generate) then jumping to step 2 of the following procedure.
+<!-- ? I noticed the strapi generate CLI generates an empty router and not a call to createCoreRouter, is it intended ? -->
```

Comment:
Yes, an empty router like:

```js
const { createCoreRouter } = require('@strapi/strapi').factories;

module.exports = createCoreRouter('api::test.test');
```

Is all the user needs **if they didn't create any custom routes or customize the default routes**